### PR TITLE
QnAMakerDialog adaption SDK 480

### PR DESCRIPTION
**Pull request by vipeketi**
Fixes #

## Proposed Changes

  - 
  -
  -

### CODE REVIEW

File: samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js [hunk 1]
```diff
@@ -31,230 +40,52 @@ const PreviousQnAId = 'prevQnAId';
...
-        this.addDialog(new WaterfallDialog(QNAMAKER_DIALOG, [
-            this.callGenerateAnswerAsync.bind(this),
-            this.callTrain.bind(this),
-            this.checkForMultiTurnPrompt.bind(this),
-            this.displayQnAResult.bind(this)
-        ]));
-
-        this.initialDialogId = QNAMAKER_DIALOG;
+    constructor(knowledgebaseId, authkey, host) {
+        console.log('qnamkaerbasedialog'+ knowledgebaseId+' '+authkey+' '+host)
```
Comment (trojenguri) [inline samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js]: Please remove all the console.logs

File: samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js [hunk 2]
```diff
@@ -31,230 +40,52 @@ const PreviousQnAId = 'prevQnAId';
...
-    * @param {WaterfallStepContext} stepContext contextual information for the current step being executed.
-    */
-    async callGenerateAnswerAsync(stepContext) {
-        // Default QnAMakerOptions
-        var qnaMakerOptions = {
-            scoreThreshold: DefaultThreshold,
-            top: DefaultTopN,
-            context: {},
-            qnaId: -1
+    async GetQnAMakerOptions(dc)
```
Comment (trojenguri) [inline samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js]: This is not async function

File: samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js [hunk 3]
```diff
@@ -31,230 +40,52 @@ const PreviousQnAId = 'prevQnAId';
...
-            this.displayQnAResult.bind(this)
-        ]));
-
-        this.initialDialogId = QNAMAKER_DIALOG;
+    constructor(knowledgebaseId, authkey, host) {
+        console.log('qnamkaerbasedialog'+ knowledgebaseId+' '+authkey+' '+host)
+        var noAnswer = ActivityFactory.DefaultNoAnswer;
+        var strictFilters = null;
+        super(knowledgebaseId, authkey, host, noAnswer, 0.3, 'Did you mean:','None of the above.',
+        3, ActivityFactory.cardNoMatchResponse, strictFilters, QNAMAKER_BASE_DIALOG);
```
Comment (trojenguri) [inline samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js]: Make these values as const.

File: samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js [hunk 4]
```diff
@@ -31,230 +40,52 @@ const PreviousQnAId = 'prevQnAId';
...
-            context: {},
-            qnaId: -1
+    async GetQnAMakerOptions(dc)
+    {
+        console.log('GetQnAMakerOptions')
+        return new QnAMakerOptions
+        {
+            ScoreThreshold = DefaultThreshold,
+            Top = DefaultTopN,
+            QnAId = 0,
```
Comment (trojenguri) [inline samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js]: Can you please verify that this function is overriding base class's function. 
I am not sure how javascript implements overrides but in base class' function definition, no where it is mentioned that this particular function can be overridden.

File: samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js [hunk 5]
```diff
@@ -31,230 +40,52 @@ const PreviousQnAId = 'prevQnAId';
...
-    async callTrain(stepContext) {
-        var trainResponses = stepContext.values[QnAData];
-        var currentQuery = stepContext.values[CurrentQuery];
-
-        var reply = stepContext.context.activity.text;
-
-        var dialogOptions = getDialogOptionsValue(stepContext);
-        var qnaDialogResponseOptions = dialogOptions[QnADialogResponseOptions];
+    async  GetQnAResponseOptions(dialogContext)
+    {
```
Comment (trojenguri) [inline samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js]: Please verify same for this function also.

File: samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js [hunk 6]
```diff
@@ -31,230 +40,52 @@ const PreviousQnAId = 'prevQnAId';
 const QNAMAKER_DIALOG = 'qnamaker-dialog';
```
Comment (trojenguri) [inline samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js]: This line can be removed

File: samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js (hunk 1 above)
Comment (vipeketi) [reply samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js]: Removed all console logs

File: samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js (hunk 2 above)
Comment (vipeketi) [reply samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js]: removed unused function

File: samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js (hunk 3 above)
Comment (vipeketi) [reply samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js]: Used constants instead of values

**Pull request merged**